### DISCUSSION TITLE
Snap name different from distribution name?

### DISCUSSION THREAD

**Discussion by mthmulders** [jreleaser/jreleaser#588] [Q&A] [answered]
I'm trying to publish my software using Snap, but I can't register my preferred name with on [snapcraft.io](https://snapcraft.io). So I took a different Snap name there.

The snapcraft.yml that JReleaser created contains `name: {{distributionName}}`. Now `distributionName` is the preferred name but not the one I was able to claim on [snapcraft.io](https://snapcraft.io). For now, I've [monkey-patched] the snapcraft.yml file and it seems to work from Snapcraft point-of-view, but I'd rather not do that every time.

From a JReleaser point-of-view, would it be advisable to create a second distribution, with the "alternative" name, and only bind the Snap release to it - keeping the Homebrew and other releases bound to the "preferred" name?

Edit: as an aside, I am trying to get the preferred name with Snapcraft anyway, but my hopes aren't that high...

**Comment by aalmiray** [comment, marked as answer]
If the only change from the standard template is `{{distributionName}}` then you have two options:
- create a second distribution with the correct name (a bit of a code smell if it's just for the distribution name, I should fix this).
- create your own template using `jreleaser template -p snap -b . -d <currentDistributionName>` then edit the generated file in `src/jreleaser/teemplates`.

Homebrew allows the formula name to differ from the distribution name, I think Snap should have a similar setting, defaulting to `distributionName` if left unspecified.

**Reply by mthmulders** [reply]
Thanks for the ideas, Andres. For now, I think I'll stick with the template.

If you think the feature to have the Snap name different from the distribution name isn't too hard to implement (just as for Homebrew), I might be able to try and see if I could contribute. Would it make sense to open a separate ticket for that?

**Reply by mthmulders** [reply]
Another question that comes to mind... Imagine I have a custom template for e.g. Snap (or Homebrew, or...) and the one that ships with JReleaser would change, adding new features or fixing bugs.... How would I notice?

**Reply by aalmiray** [reply]
Thanks Maarten, it wasn't really that difficult to implement https://github.com/jreleaser/jreleaser/issues/589

Now for your other question, there's no option at the moment to determine how different your custom template is from the ones bundled with the tool other that generating the template once again and manually comparing against a copy of yours.

Perhaps this could be added as an option to the existing `template` command.

**Reply by mthmulders** [reply]
As a first step in answering that second question, may I suggest the `template` command to insert a comment along the lines of "generated using jreleaser vXXX"? It would at least notify people the template may be a lot older than the version of JRelease they're currently using.

Same offer as before: I'm willing to give it a shot myself and contribute back :-)

**Reply by aalmiray** [reply]
The idea of using comments is a good one to begin with. An automated check may be useful later. All templates reside inside the `jreleaser-templates` project of you want to have a look. I believe the chocolatey, macports, and spec templates require the 1st line to have a special comment but we should be good by adding more comments.
